packages/contracts/hardhat.config.ts

```diff
@@ -117,178 +116,157 @@ const config: HardhatUserConfig = {
-          apiURL: "https://gnosis-chiado.blockscout.com/api",
+          apiURL: `https://api.etherscan.io/v2/api?chainid=${EChainId.GnosisChiado}`,
           browserURL: "https://gnosis-chiado.blockscout.com/",
         },
       },
       {
         network: ESupportedChains.Polygon,
         chainId: EChainId.Polygon,
         urls: {
-          apiURL: "https://polygonscan.com/api",
+          apiURL: `https://api.etherscan.io/v2/api?chainid=${EChainId.Polygon}`,
           browserURL: "https://polygonscan.com/",
         },
       },
       {
         network: ESupportedChains.PolygonAmoy,
         chainId: EChainId.PolygonAmoy,
         urls: {
-          apiURL: "https://amoy.polygonscan.com/api",
+          apiURL: `https://api.etherscan.io/v2/api?chainid=${EChainId.PolygonAmoy}`,
           browserURL: "https://amoy.polygonscan.com/",
         },
       },
       {
         network: ESupportedChains.Linea,
         chainId: EChainId.Linea,
         urls: {
-          apiURL: "https://api.lineascan.build/api",
+          apiURL: `https://api.etherscan.io/v2/api?chainid=${EChainId.Linea}`,
           browserURL: "https://lineascan.build/",
         },
       },
       {
         network: ESupportedChains.LineaSepolia,
         chainId: EChainId.LineaSepolia,
         urls: {
-          apiURL: "https://api-sepolia.lineascan.build/api",
+          apiURL: `https://api.etherscan.io/v2/api?chainid=${EChainId.LineaSepolia}`,
           browserURL: "https://sepolia.lineascan.build/",
         },
       },
       {
         network: ESupportedChains.ZkSyncEra,
         chainId: EChainId.ZkSyncEra,
         urls: {
-          apiURL: "https://zksync2-mainnet-explorer.zksync.io/contract_verification",
+          apiURL: `https://api.etherscan.io/v2/api?chainid=${EChainId.ZkSyncEra}`,
           browserURL: "https://explorer.zksync.io",
         },
       },
       {
         network: ESupportedChains.ZkSyncSepolia,
         chainId: EChainId.ZkSyncSepolia,
         urls: {
-          apiURL: "https://explorer.sepolia.era.zksync.dev/contract_verification",
+          apiURL: `https://api.etherscan.io/v2/api?chainid=${EChainId.ZkSyncSepolia}`,
           browserURL: "https://sepolia.explorer.zksync.io",
         },
       },
       {
         network: ESupportedChains.PolygonZkEvm,
         chainId: EChainId.PolygonZkEvm,
         urls: {
-          apiURL: "https://api-zkevm.polygonscan.com/api",
+          apiURL: `https://api.etherscan.io/v2/api?chainid=${EChainId.PolygonZkEvm}`,
           browserURL: "https://zkevm.polygonscan.com/",
         },
       },
       {
         network: ESupportedChains.PolygonCardonaZkEvm,
         chainId: EChainId.PolygonCardonaZkEvm,
         urls: {
-          apiURL: "https://api-cardona-zkevm.polygonscan.com/api",
+          apiURL: `https://api.etherscan.io/v2/api?chainid=${EChainId.PolygonCardonaZkEvm}`,
           browserURL: "https://docs.polygonscan.com/cardona-polygon-zkevm",
         },
```

Comment:
does eth mainnet need adding here?

Reply:
I guess it wasn't there before so unsure if it should be

Reply:
It's supported inside the plugin, so no. These changes are needed for custom chains.